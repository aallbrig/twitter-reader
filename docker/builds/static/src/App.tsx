import * as React from 'react';
import './App.css';

const logo = require('./logo.svg');

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Twitter Reader for Target Twitter Account</h2>
        </div>
        <p className="App-intro">
          (Placeholder for filter input field) <br />
          (Placeholder for recent 10 tweets)
        </p>
      </div>
    );
  }
}

export default App;
