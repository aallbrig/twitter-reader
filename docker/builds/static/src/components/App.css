.App-logo {
  animation: App-logo-spin infinite 10s linear;
  height: 80px;
}

.App-header {
  text-align: center;
  background-color: #222;
  height: 165px;
  padding: 20px;
  margin-bottom: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
