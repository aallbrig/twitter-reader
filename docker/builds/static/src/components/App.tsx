import * as React from 'react';
import { curry } from 'lodash';
import { Grid, Row, Col, Popover, OverlayTrigger } from 'react-bootstrap';
import { IntlProvider } from 'react-intl';
import './App.css';
import TweetPanel from './TweetPanel';
import FilterTweetInput from './FilterTweetInput';
import InformationModal from './InformationModal';
import { FilterableTweet, Tweet } from '../types';

const logo = require('./logo.png');

export interface Props {
  highlightedWord: string;
  showModal: boolean;
  rawTweets: Tweet[];
  tweets: FilterableTweet[];
  getRecentTweets: () => void;
  dismissModal: () => void;
  filterTweets: (tweets: Tweet[], filterBy: string) => void;
}

export interface State {
  shake: boolean;
}

class App extends React.Component<Props, State> {
  state = {
    shake: true
  };
  componentWillMount() {
    const {getRecentTweets} = this.props;
    getRecentTweets();
    setTimeout(() => this.setState({ shake: false }), 800);
    setInterval(
      () => {
        this.setState(
          () => ({
            shake: true
          }),
          () => {
            getRecentTweets();
            setTimeout(() => this.setState({ shake: false }), 800);
          }
        );
      },
      10 * 1000
    );
  }
  render() {
    const {
      highlightedWord, tweets, rawTweets,
      filterTweets, showModal, dismissModal
    } = this.props;
    const { shake } = this.state;
    return (
      <IntlProvider locale="en">
        <div className="App">
          <div className="App-header">
            <OverlayTrigger
              placement="bottom"
              overlay={(
                <Popover id="logo-tooltip">
                  Every time this shakes, a new "get recent tweets from target user" request is sent.
                </Popover>
              )}
              trigger={['click', 'hover', 'focus']}
              delayHide={2000}
            >
              <img src={logo} className={`App-logo${shake ? ' shake-chunk shake-freeze' : ''}`} alt="logo" />
            </OverlayTrigger>
            <h1 className="h4" style={{marginTop: 22}}>Recent Tweets from @Salesforce</h1>
          </div>
          <Grid>
            <Row>
              <Col xs={12}>
                <FilterTweetInput onKeyDown={curry(filterTweets)(rawTweets)}/>
                <OverlayTrigger
                  placement="bottom"
                  overlay={(
                    <Popover id="logo-tooltip">
                      {`REST web server's HTTP cache clears after approximately `}
                      {'one minute.  HTTP cache is used here to reduce my application\'s use '}
                      {'of Twitter\'s API.'}
                    </Popover>
                  )}
                  trigger={['click', 'hover', 'focus']}
                  delayHide={2000}
                >
                  <span>
                    <i className="glyphicon glyphicon-repeat Cache-Reset" />
                    {' Time until REST server cache clears.'}
                  </span>
                </OverlayTrigger>
                <br />
                <OverlayTrigger
                  placement="right"
                  overlay={(
                    <Popover id="logo-tooltip">
                      {'For comparison.'}
                    </Popover>
                  )}
                  trigger={['click', 'hover', 'focus']}
                  delayHide={2000}
                >
                  <span>
                    <i className="glyphicon glyphicon-repeat Second-Interval" />
                    {' One second timer'}
                  </span>
                </OverlayTrigger>
              </Col>
            </Row>
            <Row>
              <Col xs={12}>
                {tweets
                  .filter(({ disabled }) => !disabled)
                  .map((tweet, index) =>
                    <TweetPanel
                      key={`tweet-${tweet.id}`}
                      {...{
                        tweet,
                        index: index + 1,  // Human readable
                        highlightedWord
                      }}
                    />)}
              </Col>
            </Row>
          </Grid>
          <InformationModal showModal={showModal} dismissModal={dismissModal} />
        </div>
      </IntlProvider>
    );
  }
}

export default App;
