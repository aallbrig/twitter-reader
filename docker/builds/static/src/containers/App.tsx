import App from '../components/App';
import * as actions from '../actions/';
import { StoreState } from '../types/index';
import { connect, Dispatch } from 'react-redux';
import { Tweet } from '../types';

export function mapStateToProps({
  filterTweets: {
    filterBy: highlightedWord, // alias for UI purposes
    filteredTweets: tweets
  },
  tweets: { tweets: rawTweets }
}: StoreState) {
    return {
      tweets,
      rawTweets,
      highlightedWord
    };
  }
  
export function mapDispatchToProps(dispatch: Dispatch<actions.TweetAction>) {
    return {
      getRecentTweets: () => dispatch(actions.getRecentTweets()),
      filterTweets: (tweets: Tweet[], value: string) => dispatch(actions.filterTweets(tweets, value))
    };
  }

export default (connect(
  mapStateToProps,
  mapDispatchToProps
) as any)(App);
