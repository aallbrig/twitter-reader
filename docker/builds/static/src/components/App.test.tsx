import * as React from 'react';
import App from './App';
import { create } from 'react-test-renderer';
import { noop } from 'lodash';

it('App renders correctly', () => {
  const tree = create(
    <App
        highlightedWord="test"
        showModal={false}
        rawTweets={[]}
        tweets={[]}
        getRecentTweets={noop}
        dismissModal={noop}
        filterTweets={noop}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});