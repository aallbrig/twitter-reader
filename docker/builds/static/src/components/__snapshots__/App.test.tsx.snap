// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo shake-chunk shake-freeze"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      src="logo.png"
    />
    <h1
      className="h4"
      style={
        Object {
          "marginTop": 22,
        }
      }
    >
      Recent Tweets from @Salesforce
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <div
          className="well well-lg"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "position": "absolute",
                "right": 10,
                "top": 10,
              }
            }
          >
            <span
              className="badge"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              ?
            </span>
          </div>
          <div
            className="form-group"
          >
            <label
              className="control-label"
              htmlFor={undefined}
            >
              Tweet Content Filter & Highlight
            </label>
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Highlight text in tweet content"
              type="text"
            />
          </div>
        </div>
        <span
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <i
            className="glyphicon glyphicon-repeat Cache-Reset"
          />
           Time until REST server cache clears.
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      />
    </div>
  </div>
</div>
`;
